### ISSUE TITLE
Add power button support

### ISSUE THREAD

**Issue by gitchomik** [jernejsk/OpenELEC-OPi2#12]
More info in threads
http://www.orangepi.org/orangepibbsen/forum.php?mod=viewthread&tid=940
and
http://www.orangepi.org/orangepibbsen/forum.php?mod=viewthread&tid=817

Regards !

**Comment by jernejsk** [comment]
If somebody creates good pull request or patch I will consider it.

**Comment by warik-zp** [comment]
I found a solution:
In script.bin:
[wakeup_src_para]
wakeup_src_bt = port:PL03<6><default><default><0>

[gpio_power_key]
key_used = 1
key_io = port:PL03<6><default><default><0>

Code screenshot: https://imgur.com/WErESa8

Kernel with option:
System Type -> Homlet platform bootup extend
Device Drivers -> Input device support -> Keyboards -> sunxi IR RX support
Device Drivers -> Input device support -> Keyboards -> gpio key for H3 prototype

**Comment by jernejsk** [comment]
sunxi IR RX is not strictly needed, only for turning it on/off with IR remote? I will port suspend/resume support to new IR driver in the future, I don't intent to go back to NEC only driver.

**Comment by warik-zp** [comment]
I'm talking about the power button. If use this option, we can add support power button?

**Comment by jernejsk** [comment]
I don't see logical connection between IR driver and power button. I also checked this IR driver code and still can't see the connection.

**Comment by warik-zp** [comment]
Oh. Sorry. This method add solution power on from IR and button.

**Comment by jernejsk** [comment]
Can you check if power button now works? Don't forget to clear u-boot first so script.bin gets recompiled.

**Comment by warik-zp** [comment]
Oaky. I'll try and let you know.

**Comment by galeots** [comment]
fix power button!?

**Comment by jernejsk** [comment]
Sorry, no. I will fix this when I will have nothing else to do. I spent too much time for this already.

**Comment by galeots** [comment]
Once solved this program is complete!

**Comment by llonca13** [comment]
@jernej thanks for everything

To be honest i dont mind this power button/ir bug,if you can fox it great if no ill just leave it as a 0-24 working htpc/mysql library server :)

**Comment by jernejsk** [comment]
I highly suspect that this bug resides somewhere in graphics stack, but I have no idea how to locate it. I will first try to fix some issues in Kodi and later try to fix it again.

**Comment by llonca13** [comment]
Hvala :)

**Comment by llonca13** [comment]
You fixed itttt :)
I tried the new version(25 april) and it works..i dont know for the ir because i have orangepi one..but the button works perfectly..shutdown in kodi puts it in suspended state(with the red light glowing) which is in my opinion alot better then turning it off :)
And i also noticed the system info show correct amount of memory now

Thank you soooo much jernej
I think this issue can be closed now 
Thank you

**Comment by jernejsk** [comment]
Are you pretty sure? It seems that it doesn't work on my OPi2. I need more reports...

**Comment by llonca13** [comment]
yes im sureee :dancers:  :)
sorry to hear about your Opi2 but on Opi One you fixed the problem.

heres video : 
https://www.youtube.com/watch?v=wu8Ong0On2Q

**Comment by jernejsk** [comment]
Good for you, but I won't close it until it works reliably for everyone.

**Comment by galeots** [comment]
video not work! is corrupt

**Comment by jernejsk** [comment]
@galeots or anyone else,
can you confirm that suspend/resume using button almost works IF there is NO USB mouse/keyboard connected?

**Comment by galeots** [comment]
@jernejsk  Tomorrow I try to compile if I have time I'll tell you

**Comment by galeots** [comment]
@jernejsk The suspension goes without any connected USB devices. sata no longer working in the new version

**Comment by jernejsk** [comment]
@galeots,
SATA can be fixed, just wait few days till I get back to my OPi.

In the mean time, I think I found the issue with USB. Please check issue #41 and test opiplus.fex change described there.

**Comment by galeots** [comment]
@jernejsk you can make sure that when goes in suspension turns off the fan !?

**Comment by jernejsk** [comment]
FAN is not official addon, so I wouldn't put this in image. I can help you with that on the forum. This is a bit off topic.

**Comment by galeots** [comment]
@jernejsk thx

**Comment by jernejsk** [comment]
@galeots Did you test fix in #41 ?

**Comment by galeots** [comment]
@jernejsk not work fix boot erase PG11 for sata

**Comment by galeots** [comment]
@jernejsk how are the works for fix the power button?

**Comment by jernejsk** [comment]
It works as long as you don't use USB hub with attached devices. This unfortunatelly means that you should not use USB devices on OPi2 and OPiPlus. You can use IR remote for a workaround.

**Comment by galeots** [comment]
@jernejsk but the bug has what is due !?

**Comment by jernejsk** [comment]
Sorry, I don't understand what you mean. Button works. The only issue is USB driver which hangs at suspend if there is hub with device connected.

**Comment by galeots** [comment]
@jernejsk the bug is due because usb does not suspend !?

**Comment by jernejsk** [comment]
Yes, I will close it once there will be no known problems with suspend / resume cycle.

**Comment by galeots** [comment]
@jernejsk however if you do leave without USB peripherals work suspension! if you take away and put USB peripherals and suspend doesn't go

**Comment by jernejsk** [comment]
That is what I'm saying. You have Plus. If you would have PC or One, you wouldn't have any of those problem.
